src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -1063,6 +1068,18 @@ - (void)_handleReportEnd
         _clustersToPersist = nil;
     }
 
+    // After the handling of the report, if we detected a device configuration change, notify the delegate
+    // of the same.
+    if (_deviceConfigurationChanged) {
```

Comment:
You'll want to reset `_deviceConfigurationChanged` to `NO` after this.

Reply:
done.